emgapianns/management/commands/clean_study.py

```diff
@@ -18,105 +18,151 @@
-
-    groups = {
-        "ITS": [
-            "Taxonomic analysis ITS",
-            "Taxonomic analysis ITSoneDB",
-            "Taxonomic analysis UNITE"
-        ],
-        "rRNA": [
-            "Taxonomic analysis SSU rRNA",
-            "Taxonomic analysis LSU rRNA"
-        ]
-    }
+    help = (
+        "Remove the results download files and/or taxonomic annotations for a study analyses. "
+        "Mongo annotations are re-feshed with the exception of the taxonomic ones "
+        "(hence the option on this command)"
+    )
 
     def add_arguments(self, parser):
+        # we need to get all the Download Groups
+        dgroup_slugs = [_slugify(g.group_type) for g in DownloadGroupType.objects.all()]
+        dgroup_slugs.append("all")
+
         super(Command, self).add_arguments(parser)
-        parser.add_argument("accession", action='store',
-                            type=str, help="Study accession")
-        parser.add_argument("download_group", help="Download group", type=str,
-                            choices=[g for g in self.groups.keys()])
+        parser.add_argument("-a", "--accession", required=True, help="Study accession")
+        parser.add_argument(
+            "-m",
+            "--mongo_tax",
+            help="Delete mongo taxonomic annotations",
+            action="store_true",
+        )
+        parser.add_argument(
+            "-g",
+            "--group",
+            nargs="+",
+            required=True,
+            help="DB Files ONLY, download groups options: %(choices)s",
+            metavar="GROUP",
+            type=str,
+            choices=dgroup_slugs,
+        )
         parser.add_argument(
-            '--pipeline',
-            help='Pipeline version',
-            action='store',
-            dest='pipeline',
-            choices=[
-                '1.0',
-                '2.0',
-                '3.0',
-                '4.0',
-                '4.1',
-                '5.0'],
-            default='5.0'
+            "--pipeline",
+            help="Pipeline version",
+            action="store",
+            dest="pipeline",
+            choices=["1.0", "2.0", "3.0", "4.0", "4.1", "5.0"],
+            default="5.0",
         )
 
     def handle(self, *args, **options):
-
+        """Execute the command"""
         pipeline_version = options.get("pipeline")
         study_id = int(options.get("accession").replace("MGYS", ""))
-        study = Study.objects.get(pk=study_id)
 
-        download_group = options.get("download_group")
-        selected_groups = self.groups[download_group]
-        groups = DownloadGroupType.objects.filter(group_type__in=selected_groups)
+        try:
+            study = Study.objects.get(pk=study_id)
+        except Study.DoesNotExist:
+            logger.error("Study " + str(study_id) + " doesn't exist")
+            return
+
+        download_groups = options.get("group")
+
+        selected_groups = []
+        if "all" in download_groups:
```

Comment:
will you ever have 'all'? choices for group are limited to dgroup_slugs?

Reply:
Yes you can, I've manually added `all` to dgroup_slugs